core/thread/inc/TPool.h

```diff
@@ -0,0 +1,204 @@
+#ifndef TPOOL_H_
+#define TPOOL_H_
```

Comment:
Can you use ROOT-style include guards? Makes it much easier to understand where those CPP defines come from...